2-extra/6-cheap-diner.js

```diff
@@ -19,7 +19,35 @@ chosenMeal(emptyArray)
 Should give the answer "Nothing :("
 **/
 
-function chooseMeal(mealArray) {}
+function chooseMeal(mealArray) {
+  let small = Infinity, secondSmall = Infinity;
+
+  if (mealArray.length === 0){
+    return "Nothing :("
+  }
+
+
+
+  for (let meal of mealArray){
+    if (mealArray.length === 0){
+      return "Nothing :("
+    }else if (mealArray.length === 1){
+      return meal.name
+    }else if (meal.price < small){
```

Comment:
Could you explain the logic here with the < small condition for each meal of mealArray? does this logic have a name?